src/main/java/com/faforever/client/remote/FafService.java

```diff
@@ -466,11 +466,15 @@ public void restoreGameSession(int id) {
   }
 
   @Async
-  public CompletableFuture<Tuple<List<MapBean>, Integer>> getLadder1v1MapsWithPageCount(int count, int page) {
-    Tuple<List<Ladder1v1Map>, java.util.Map<String, ?>> tuple = fafApiAccessor.getLadder1v1MapsWithMeta(count, page);
+  public CompletableFuture<Tuple<List<MapBean>, Integer>> getMatchmakerMapsWithPageCount(int matchmakerQueueId, int count, int page) {
+    // fixme This uses the count for the number of map pools instead of the number of maps
+    Tuple<List<MatchmakerQueueMapPool>, java.util.Map<String, ?>> tuple = fafApiAccessor.getMatchMakerPoolsWithMeta(matchmakerQueueId, count, page);
     return CompletableFuture.completedFuture(new Tuple<>(tuple.getFirst()
         .parallelStream()
-        .map(ladder1v1Map -> MapBean.fromMapVersionDto(ladder1v1Map.getMapVersion()))
+        .map(MatchmakerQueueMapPool::getMapPool)
+        .flatMap(mapPool -> mapPool.getMapVersions().stream())
+        .distinct()
+        .map(MapBean::fromMapVersionDto)
         .collect(toList()),
```

Comment:
If you want to keep this structure and implement the pagination you would need to do it here with some .skip and .limit functions but that wouldn't work with the parallelStream

Reply:
I made it work! Is there a more elegant way to find out the size of the stream than what I did? It feels a bit clunky.

Reply:
Unfortunately not that I am aware of as .count() is a terminal operation on a stream. Also your total pages might be off by 1 it might be good to get the ceiling of that value.